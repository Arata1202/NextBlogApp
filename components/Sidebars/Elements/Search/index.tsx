import { MagnifyingGlassIcon } from '@heroicons/react/24/solid';
import SearchField from '../../../Elements/SearchField';
import styles from './index.module.css';

interface Props {
  isDarkMode?: boolean;
}

export default function Search({ isDarkMode }: Props) {
  return (
    <div
      className={`pt-8 px-4 border py-5 ${isDarkMode ? 'DarkTheme border-gray-500 placeholder:text-gray-500' : 'lightTheme border-gray-300 placeholder:text-gray-300'}`}
    >
      <h1
        className={`${styles.profile} text-2xl text-center font-semibold mb-5 flex justify-center`}
      >
        <MagnifyingGlassIcon className="h-8 w-8 mr-2" aria-hidden="true" />
        キーワードで探す
      </h1>
      <SearchField isDarkMode={isDarkMode} />
    </div>
  );
}
