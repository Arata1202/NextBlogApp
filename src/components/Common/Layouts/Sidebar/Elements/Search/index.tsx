import { useTheme } from 'next-themes';
import { MagnifyingGlassIcon } from '@heroicons/react/24/solid';
import SearchField from '@/components/Elements/SearchField';

export default function Search() {
  const { theme } = useTheme();

  return (
    <>
      <div className={`pt-8 px-4 border py-5 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}>
        <div className={`text-2xl text-center font-semibold mb-5 flex justify-center`}>
          <MagnifyingGlassIcon className="h-8 w-8 mr-2" />
          キーワードで探す
        </div>
        <SearchField />
      </div>
    </>
  );
}
