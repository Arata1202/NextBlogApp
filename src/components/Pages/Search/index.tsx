import { Article } from '@/libs/microcms';
import PageHeading from '@/components/Common/PageHeading';
import Pagination from '@/components/Common/Pagination';
import ArticleList from '@/components/Common/ArticleList';
import Sidebar from '@/components/Common/Layouts/Sidebar';
import Display from '@/components/Common/Adsense/Display';

type Props = {
  articles: Article[];
  keyword?: string;
  current?: number;
  totalCount: number;
  recentArticles?: Article[];
};

export default function SearchPage({
  articles,
  keyword,
  totalCount,
  recentArticles,
  current,
}: Props) {
  return (
    <>
      <PageHeading keyword={keyword} />
      <ArticleList articles={articles} recentArticles={recentArticles} />
      <Pagination totalCount={totalCount} current={current} basePath={`/search`} q={keyword} />
      <Sidebar recentArticles={recentArticles} mobile={true} />
      <div className="mt-5">
        <Display slot="5969933704" />
      </div>
    </>
  );
}
