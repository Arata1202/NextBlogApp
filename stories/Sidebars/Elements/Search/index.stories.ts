import type { Meta, StoryObj } from '@storybook/react';
import Search from '@/components/Sidebars/Elements/Search';

const meta: Meta<typeof Search> = {
  title: 'Sidebars/Elements/Search',
  component: Search,
  parameters: {
    layout: 'fullscreen',
  },
  argTypes: {
    isDarkMode: { control: 'boolean' },
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    isDarkMode: false,
  },
};
