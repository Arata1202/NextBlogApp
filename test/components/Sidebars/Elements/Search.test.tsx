import { render } from '@testing-library/react';
import Search from '@/components/Sidebars/Elements/Search';

describe('Search', () => {
  test('スナップショット（Search）', () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });
});
