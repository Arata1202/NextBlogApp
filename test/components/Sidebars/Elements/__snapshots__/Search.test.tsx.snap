// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search スナップショット（Search） 1`] = `
<DocumentFragment>
  <div
    class="pt-8 px-4 border py-5 LightTheme"
  >
    <h1
      class="profile text-2xl text-center font-semibold mb-5 flex justify-center"
    >
      <svg
        aria-hidden="true"
        class="h-8 w-8 mr-2"
        data-slot="icon"
        fill="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10.5 3.75a6.75 6.75 0 1 0 0 13.5 6.75 6.75 0 0 0 0-13.5ZM2.25 10.5a8.25 8.25 0 1 1 14.59 5.28l4.69 4.69a.75.75 0 1 1-1.06 1.06l-4.69-4.69A8.25 8.25 0 0 1 2.25 10.5Z"
          fill-rule="evenodd"
        />
      </svg>
      キーワードで探す
    </h1>
  </div>
</DocumentFragment>
`;
